basic-services/ReadPropertyResult.cs

```diff
@@ -39,42 +40,130 @@ public struct ReadPropertyResult
+                    BooleanValue = Convert.ToBoolean(NumericValue);
+                    break;
                 case JTokenType.String:
-                    // todo
+                    NumericValue = 0;
+                    StringValue = token.Value<string>();
+                    break;
                 case JTokenType.Array:
-                    // todo
+                    ReadArray(token);
+                    break;
                 case JTokenType.Boolean:
-                    // todo
+                    if ((bool)(token) == true) {
+                        NumericValue = 1;
+                        StringValue = "True";
+                        BooleanValue = true;
+                    } else {
+                        NumericValue = 0;
+                        StringValue = "False";
+                        BooleanValue = false;
+                    }
+                    break;
+                case JTokenType.Object:
+                    ReadObject(token);
+                    break;
                 default:
-                    StringValue = null;
-                    NumericValue = 0;
-                    ArrayValue = null;
-                    BooleanValue = false;
+                    NumericValue = 1;
+                    StringValue = "Unsupported Data Type";
                     break;
             }
-            
         }
 
-    }
-
-    
+        /// <summary>Parses a JArray and adds (string, double) pairs to an array based on JToken type.</summary>
+        internal void ReadArray(JToken token) {
+            JArray arr = JArray.Parse(token.ToString());
+            ArrayValue = new (string, double, bool)[arr.Count];
+            int index = 0;
+            foreach(var item in arr.Children())
+            {
+                double value = 0;
+                switch (item.Type)
+                {
+                    case JTokenType.Integer:
+                        value = (double) item.Value<double>();
+                        ArrayValue[index] = (value.ToString(), value, Convert.ToBoolean(NumericValue));
+                        break;
+                    case JTokenType.Float:
+                        value = (double) item.Value<double>();
+                        ArrayValue[index] = (value.ToString(), value, Convert.ToBoolean(NumericValue));
+                        break;
+                    case JTokenType.String:
+                        ArrayValue[index] = (item.Value<string>(), value, false);
+                        break;
+                    default:
+                        ArrayValue[index] = ("Unsupported Data Type", 1, false);
+                        break;
+                }
+                index++;
+            }
+            NumericValue = 0;
+            StringValue = "Array";
+        }
 
+        /// <summary>Searches the JObject for reliability and priority fields and finds a value to use.</summary>
+        internal void ReadObject(JToken token) {
```

Comment:
Did you write this to process `presentValue`? I assume so because we don't support object data types (except in this case). If so, we can make this more explicit and call it `ProcessPresentValue`. 

Note: at 11.0 `presentValue` will no longer be a special case. It'll be handled like any other attribute.

Reply:
I did this with presentValue in mind. I wasn't sure if there was another attribute where this happened.

Reply:
No, it's an unfortunate design decision that's corrected at 11.